src/controllers/FacetSearchController.ts

```diff
@@ -5,14 +5,16 @@ import { IFacetSearchResponse } from '../rest/Facet/FacetSearchResponse';
 export class FacetSearchController {
   constructor(private facet: DynamicFacet) {}
 
-  public search(query?: string): Promise<IFacetSearchResponse> {
+  public search(terms?: string): Promise<IFacetSearchResponse> {
+    const optionalLeadingWildcard = this.facet.options.useLeadingWildcardInFacetSearch ? '*' : '';
+
     const request: IFacetSearchRequest = {
       field: this.facet.fieldName,
       numberOfValues: this.facet.options.numberOfValues,
       ignoreValues: this.facet.values.allValues,
       captions: this.facet.options.valueCaption,
       searchContext: this.facet.queryController.getLastQuery(),
-      query
+      query: `${optionalLeadingWildcard}${terms}*`
```

Comment:
What happens if someone searches with `terms` that contains `*` ?

Reply:
Good question, I'll ask but my guest that it's the API that's going to manage that. If we need to do something on our side I'll manage it in a further PR